shared-libs/transitions/src/lib/utils.js

```diff
@@ -229,6 +242,34 @@ module.exports = {
 
   isXFormReport: doc => doc && doc.type === 'data_record' && doc.content_type === 'xml',
 
+  /**
+   * Validates if a cron expression is valid and is active within an offset timeframe at the current time.
+   * @param {string} exp cron expression without seconds "* * * * *"
+   * @param {number} frame timeframe at which the cron expression should still be valid
+   * @returns true if valid and within timeframe
+   */
+  isWithinTimeFrame: (exp, frame = 0) => {
+    if (!isValidCronExpression(exp)) {
+      throw new Error(`isWithinTimeFrame: Invalid cron expression "${exp}"`);
+    }
+    
+    const currentTime = Date.now();
+    const parsedCron = later.parse.cron(exp);
+    const nextDueTime = later.schedule(parsedCron).next().getTime();
+    const prevDueTime = later.schedule(parsedCron).prev().getTime();
+
+    const lowerBoundDueTime = nextDueTime - frame;
+    const upperBoundDueTime = prevDueTime + frame;
+    return currentTime >= lowerBoundDueTime || currentTime <= upperBoundDueTime;
```

Comment:
I'm trying to understand how this works. 
Suppose that my cron is `* 6 * * *` and the current date time is `6:49:20`.
What the code does right now: 
- computes a date which is today at 6:50:00
- computes a date which is today at 6:49:00
- we then do: `6:49:20` >= `6:50:00` || `6:49:20` <= ` 6:49:00` which obviously returns false

I believe, in this example, the function should return true, any time between `6:00` and `6:59:59` should return true.

Reply:
Not exactly, here's how it's works:

```js
_current_time = '2023-01-01 02:00:30' // time sentinel fires

exp =  '0 2 * * *' // runs on 2am precisely
frame = 60000 // 1 minute

calculation = '2023-01-01 02:01:00' >= '2023-01-01 02:00:00' || '2023-01-01 02:01:00' <= '2023-02-01 02:01:00'

returns = true // current time is within 1 minute of the expected fire time.
```